### DISCUSSION TITLE
Custom Markdown editor

### DISCUSSION THREAD

**Discussion by loranger** [filamentphp/filament#1294] [Ideas]
Hi,

It would be nice if we could extend the new [markdown editor](https://filamentadmin.com/docs/2.x/forms/fields#markdown-editor) with custom implementation such as video tag support and so on…

I don't know what's under the hood, but it would be very convenient to allow extension, such as [parsedown does](https://github.com/erusev/parsedown/wiki/Tutorial:-Create-Extensions).

**Comment by ryangjchandler** [comment]
@loranger The markdown editor currently uses a Javascript library to parse and render the Markdown as HTML - although I am open to the idea of using Commonmark on the server to generate the preview instead.

The performance might take a small hit when previewing but it won't be noticeable for the average document. Thoughts @danharrin?

**Reply by danharrin** [reply]
If we did server-rendered previews, wouldn't we need to stop deferring the `@entangle`... which would be a performance hit for lots of people.

**Reply by ryangjchandler** [reply]
It can still be deferred. The data would only be sent to the server when somebody actually clicks the preview tab. That's how the current JavaScript preview works I believe.

**Reply by pluma** [reply]
I'd like to bump this idea. The markdown editor is nice but not being able to extend it the same way Laravel's markdown implementation can be extended is a serious limitation for me. Being able to provide custom buttons and having SSR with Laravel's own markdown renderer would be a big improvement.

**Comment by Plytas** [comment]
@danharrin @ryangjchandler I see that markdown field is in fact using commonmark now. I dug around and it seems that with a few simple changes it would be possible to add extension support. Would you be open to PR for this?

**Reply by danharrin** [reply]
In v4, yes
